test/typeinfo/src/isbaseof.d

```diff
@@ -0,0 +1,35 @@
+// https://issues.dlang.org/show_bug.cgi?id=20178
```

Comment:
Can you add:

- A test with a templated interface / templated class (both `C4(T) : I!T` and `C4(T) : I!int` let's say).
- A test that it can't be called with any other typeinfo (`static assert(!__traits(compiles, ...));`)